addons/mail_plugin/controllers/mail_plugin.py

```diff
@@ -182,14 +179,25 @@ def _iap_enrich(self, domain):
                 enriched_data = {'enrichment_info': {'type': 'no_data', 'info': 'The enrichment API found no data for the email provided.'}}
         return enriched_data
 
-    def _find_existing_company(self, normalized_email):
-        sender_domain = tools.email_domain_extract(normalized_email)
-        search = sender_domain if sender_domain not in iap_tools._MAIL_DOMAIN_BLACKLIST else normalized_email
-        return request.env['res.partner'].search([('is_company', '=', True), ('email_normalized', '=ilike', '%' + search)],
-                                                 limit=1)
+    def _find_existing_company(self, email):
+        """Find the company corresponding to the given domain and its IAP cache.
 
-    def _prepare_company_values(self, company):
+        :param str domain: domain of the company we search
+        :return <res.partner>, dict: company, iap_enrich_info
+        """
+        search = self._get_search_domain(email)
+
+        partner_iap = request.env["res.partner.iap"].sudo().search([("search_domain", "=", search)], limit=1)
+
+        if partner_iap:
+            return partner_iap.partner_id
+
+        partner = request.env["res.partner"].search(
+            [("is_company", "=", True), ("email_normalized", "=ilike", "%" + search)], limit=1)
+
+        return partner
 
+    def _prepare_company_values(self, company):
```

Comment:
By the way, could we improve a bit naming of this method ? Generally it indicates we prepare values for a create; here we prepare values for plugin if I am not wrong.

Reply:
There're also
- _prepare_partner_values
- _prepare_contact_values
- _prepare_company_values

I propose `_get_company_data` (and similar thing for other method)

Reply:
Would add a suffix like "_for_plugin", to indicate this is a custom data formatting for the plugin iself. Would make ``_get_company_data_for_plugin``, what do you think ?